pychunkedgraph/backend/history.py

```diff
@@ -2,72 +2,120 @@
 import numpy as np
 import datetime
 import pandas as pd
-from networkx import DiGraph
+import networkx as nx
+import fastremap
+import requests
+import os
 
+from pychunkedgraph.backend import lineage
 from pychunkedgraph.backend.utils import column_keys
 from pychunkedgraph.backend import chunkedgraph_exceptions as cg_exceptions
 
-former_parent_col = column_keys.Hierarchy.FormerParent
-operation_id_col = column_keys.OperationLogs.OperationID
 
-class SegmentHistory(object):
-    def __init__(self, cg, root_id):
-        if not cg.is_root(root_id):
-            raise cg_exceptions.Forbidden
+def get_username_dict(user_ids, auth_token):
```

Comment:
Looks good except this part. I think it is better to pass return value of this function to `History` object. Frontend `app` module is dependent on `backend`, `get_username_dict` calling app server endpoints from inside `backend` feels like a bad circular dependency design even though it works.

`backend` components ideally should not depend on auth or user management system.

Reply:
Agreed. Fixed that